ec-europa/ecl-twig
# fix(pre-commit): Different approach for combining husky and grump-php - FRONT-960

# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

So, the logic here is:
We now have a way to combine grumphp and husky, we first execute grump-php and basing on its output we run or not the husky pre-commit script.
This way we prevent the commit from happening when there is an issue in one of our templates, the whole library is checked and not only the "staged" files which would normally happen using the pre-commit hook provided by grump-php, but the total number of files to be checked is really small, so it's not a performance issue.
The grump-en and grump-dis scripts have been replaced by node scripts and can be executed through yarn, the documentation has been updated.
To configure you environment:
`composer update`
`yarn grump-en`

Should be enough, you should now see your commit "failing" if you modify any twig template introducing a violation (like a space at the end of a line, two spaces, whatever) while if you then fix it you should see the green output by grump-php plus the checks by husky.
When it fails it should show the output of the twigcs script, unfortunately in this version is not possible to limit the output to "errors" and we receive lots of warnings that are false positive.
I briefly tested the 4.0 version of twigcs which is still in beta, it spots apparently many violations in our templates, we might soon move to that where we will also be able to set the twig version to check the code against.

This solution is using grep, hope it's not an issue for the current developers.
The changes in the site header template are the result of this implementation, grump-php was "failing" due to that.
